src/drivers/batt_smbus/batt_smbus.h

```diff
@@ -229,7 +232,7 @@ class BATT_SMBUS : public I2CSPIDriver<BATT_SMBUS>
 
 	perf_counter_t _cycle{perf_alloc(PC_ELAPSED, "batt_smbus_cycle")};
 
-	static const uint8_t MAX_NUM_OF_CELLS = 7;
+	static const uint8_t MAX_NUM_OF_CELLS = 16;
 	float _cell_voltages[MAX_NUM_OF_CELLS] {};
```

Comment:
Are there any downsides to increasing MAX_NUM_OF_CELLS? My guess is no, but we should double check because 7 seems like an odd constraint to begin with.

Reply:
7 is the max number of cells for BQ40Z80 and BQ40Z50, 16 is the max cell number for BQ78350. This constant could be removed now that there are different max cells depending on the model. It will only clip the average cell number calculation if BAT_N_CELLS is set wrong, higher than the max number of cells given by MAX_NUM_CELLS